Client/Application/AppDelegate.swift

```diff
@@ -243,9 +243,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         guard let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) else {
             return false
         }
-        if components.scheme != "firefox" && components.scheme != "firefox-x-callback" {
+
+        guard let urlTypes = NSBundle.mainBundle().objectForInfoDictionaryKey("CFBundleURLTypes") as? [AnyObject],
```

Comment:
I'm wondering whether `urlTypes.first?["CFBundleURLSchemes"]` would be safer, just in case there is nothing at index 0